src/main/java/com/brightpath/learnify/domain/auth/PermissionAccessService.java

```diff
@@ -37,11 +40,30 @@ public class PermissionAccessService {
     private final PersistentMapper persistentMapper;
     private final UuidProvider uuidProvider;
     private final UserIdentityService userIdentityService;
+    private final UserService userService;
+
+    public FullResourcePermissionModel getFullPermissionsForResource(UUID resourceId, ResourceType convertedResourceType) {
+        PermissionsAccess access = permissionAccessRepository.findById(permissionAccessId(resourceId, convertedResourceType))
+                .map(persistentMapper::asPermissionsAccess)
+                .orElseThrow(() -> new ResourceNotFoundException(convertedResourceType));
+        List<User> users = userService.getUsersByIds(access.permissions().stream().map(Permission::userId).toList());
+        Map<User, ResourceAccessEnum> userPermissions = access.permissions().stream()
+                .collect(Collectors.toMap(permission -> users.stream().filter(user -> user.id().equals(permission.userId())).findFirst().get(), Permission::resourceAccessEnum));
+        return new FullResourcePermissionModel(
+                access.resourceType(),
+                access.resourceId(),
+                access.permissionLevel(),
+                userPermissions
+        );
+    }
 
 
     // method for checking if user has access to a resource with a given id and type
     public ResourceAccessEnum getUserAccessForResource(String userId, UUID resourceId, ResourceType resourceType) {
         ResourceAccessSummary access = permissionAccessRepository.findUserAccessToResource(userId, permissionAccessId(resourceId, resourceType));
+        if (access == null) {
+            throw new ResourceNotFoundException(resourceType);
```

Comment:
is it right exception for the situation? No access found doesn't mean that the resource doesn't exists. But we can leave it as it is

Reply:
In the SQL query there is `LEFT JOIN` on user permissions, so there are few cases: 
* If there is permission record for requested user - the object is returned with all fields present (specified to single user)
* If there is no permission record for requested user - the object is returned only with basic fields (global permission level and ownerID)
* If there is no permission record AND there is no record in PermissionAccess the null is returned - I assumed it is impossible to have resource without associated permission model